BearlyHumans/squirrelville
# Final iteration of NPC dialogue

This PR improves NPC dialogue.

* Moves dialogue sentences to scriptable objects
* Different dialogue can be specified for before and after an NPC's food requirement is satisfied
* An inspector event has been to allow things to happen when dialogue finishes after the player has collected enough food

Steps to test:
1. Be on Jordan scene.
2. Speak with the big capsule NPC. Notice the dialogue it says.
3. Pick up 3 food and spit them out in the transparent food area to meet the NPC's requirements, then speak with the NPC. Notice the different dialogue.
4. Notice how the food and the food area are destroyed after the dialogue finishes.
5. Speak to the NPC. Notice the different dialogue.
6. Restart the game and repeat these steps but try putting more than the required food in the food area. Notice how only the required food is destroyed when speaking with the NPC.

https://user-images.githubusercontent.com/62001991/135579944-39658f4a-dc8c-4b54-8863-f6f526e95741.mp4
